### ISSUE TITLE
Running app.py

### ISSUE THREAD

**Issue by jainishmehta** [tanvir-ishraq/HealifyAI--LLM-based-Healthcare-System#1]
Hey man, really cool project. I was trying to do something similar and you did a great job. I am trying to run your model from your HuggingFace python file. gradio does have outputs, but not sure why I get this error

![Screenshot from 2023-11-18 21-51-48](https://github.com/tanvir-ishraq/HealifyAI--LLM-based-Healthcare-System/assets/45054638/ed1e0ca9-c35a-4e0c-bb00-be85af36248a)

**Comment by tanvir-ishraq** [comment]
@jainishmehta 
Gradio package is possibly bringing in changes. Pretty recent.

> GradioDeprecationWarning: Usage of gradio.outputs is deprecated, and will not be supported in the future, please import your components from gradio.components

Did you use/follow the requirements.txt of the project? As it is stated,
`gradio==3.34.0`
As always remember the python best practice of using virtual environment for installing packages.

**Comment by jainishmehta** [comment]
thanks :)

**Comment by jainishmehta** [comment]
I use this code on your app.py now:

import gradio as gr
from transformers import AutoTokenizer
import torch, json

from fastai.text.all import *
from blurr.text.modeling.all import *

from blurr.text.data.all import *
from fastai.vision.all import *

with open('question_labels.json', 'r') as f:
  question_dictionary = json.load(f)
que_classes = list(question_dictionary.keys())

blurr_model = load_learner('healifyLLM-stage4.pkl')
probs = blurr_model.blurr_predict('headache')[0]['probs']
values_and_probs = dict(zip(que_classes, map(float, probs)))
print(sorted(values_and_probs.items(), key=lambda x: x[1], reverse=True)[:5])


def detect_question(text):
  blurr_model = load_learner('healifyLLM-stage4.pkl')
  probs = blurr_model.blurr_predict(text)[0]['probs']
  values_and_probs = dict(zip(que_classes, map(float, probs)))
  return dict(sorted(values_and_probs.items(), key=lambda x: x[1], reverse=True))

label = gr.outputs.Label(num_top_classes=5)
iface = gr.Interface(fn=detect_question, inputs="text", outputs = label)
if __name__ == "__main__":
    iface.launch(show_api=False)   
    
    
![Screenshot from 2023-11-24 16-34-22](https://github.com/tanvir-ishraq/HealifyAI--LLM-based-Healthcare-System/assets/45054638/512615f9-f84a-4269-a4e7-618f14f8d6e2)
I get the right values from your model (so the model is not the issue)
however I keep getting error:
![Screenshot from 2023-11-24 16-35-22](https://github.com/tanvir-ishraq/HealifyAI--LLM-based-Healthcare-System/assets/45054638/61b3c0fd-174c-427d-862b-fc952939479b)

**Comment by tanvir-ishraq** [comment]
I directly deployed on huggingface where there was no issue. So, I am sure it is supposed to work locally which is what you are doing. I am sharing requirements.txt of huggingface directly here. Try installing that on a fresh virtual environment:

`gradio==3.34.0`
`gradio_client==0.2.6`
`ohmeow-blurr==1.0.5`
`torch==1.13.1`
`transformers==4.33.2 `

In the upcoming days I will check the app code locally if possible in my free time.

**Comment by jainishmehta** [comment]
Hey, I fixed it. I had to do gradio==4.8.0 and change to "
label = gr.Label(num_top_classes=5)"

**Comment by tanvir-ishraq** [comment]
That’s fantastic news! 🎉
I think I will bring a second update to the HuggingFace gradio app soon, to make it a bit more interactive. Feel free to check it out.
